Source/Core/BoogiePL.atg

```diff
@@ -42,15 +42,20 @@ public static int Parse (string/*!*/ filename, /*maybe null*/ List<string/*!*/>
   if (filename == "stdin.bpl") {
     var s = ParserHelper.Fill(Console.In, defines);
     return Parse(s, filename, out program, useBaseName);
-  } else {
-    FileStream stream = new FileStream(filename, FileMode.Open, FileAccess.Read, FileShare.Read);
-    var s = ParserHelper.Fill(stream, defines);
-    var ret = Parse(s, filename, out program, useBaseName);
-    stream.Close();
-    return ret;
+  } else
+  {
+    using (FileStream stream = new FileStream(filename, FileMode.Open, FileAccess.Read, FileShare.Read))
+    {
+      return Parse(stream, filename, defines, out program, useBaseName);
+    }
   }
 }
 
+public static int Parse(Stream stream, string/*!*/ filename, List<string> defines, out /*maybe null*/ Program program, bool useBaseName)
```

Comment:
Perhaps make the signature consistent with the above `Parse` method (default value of `useBaseName`, throws annotation).